### ISSUE TITLE
CS0105 generated for auto-generated code when using global usings with C# 10

### ISSUE THREAD

**Issue by martincostello** [dotnet/roslyn#54255]
**Version Used**: 

.NET SDK `6.0.100-preview.5.21302.13` with `<LangVersion>preview</LangVersion>`.

**Steps to Reproduce**:

1. Clone martincostello/alexa-london-travel@9d7343bedcc72cd6e5a215109430811507b0bf04
2. Remove `<NoWarn>$(NoWarn);CS0105</NoWarn>` on line 25 of `Directory.Build.props`
3. `dotnet build` from the solution root

**Expected Behavior**:

Either:

1. Tooling generating the code does not emit global usings, or;
2. Compiler ignores CS0105 warning for auto-generated code.

**Actual Behavior**:

Compiler emits CS0105 errors/warnings for duplicated using statements due to usage of global using statements.

```pwsh
C:\Coding\martincostello\alexa-london-travel [DotNet-6 ≡ +0 ~1 -0 !]> dotnet build
Microsoft (R) Build Engine version 17.0.0-preview-21302-02+018bed83d for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  The project C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\LondonTravel.Skill.EndToEndTests.csproj is using CentralPackageVersionManagement, a NuGet preview feature.
  The project C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\LondonTravel.Skill.csproj is using CentralPackageVersionManagement, a NuGet preview feature.
  The project C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.Tests\LondonTravel.Skill.Tests.csproj is using CentralPackageVersionManagement, a NuGet preview feature.
  Restored C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\LondonTravel.Skill.csproj (in 273 ms).
  Restored C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\LondonTravel.Skill.EndToEndTests.csproj (in 315 ms).
  Restored C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.Tests\LondonTravel.Skill.Tests.csproj (in 317 ms).
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\obj\Debug\net6.0\.NETCoreApp,Version=v6.0.AssemblyAttributes.cs(2,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\LondonTravel.Skill.csproj]
C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\obj\Debug\net6.0\LondonTravel.Skill.AssemblyInfo.cs(11,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\LondonTravel.Skill.csproj]
C:\Users\martin.costello\.nuget\packages\microsoft.net.test.sdk\16.10.0\build\netcoreapp2.1\Microsoft.NET.Test.Sdk.Program.cs(2,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\LondonTravel.Skill.EndToEndTests.csproj]
C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\obj\Debug\net6.0\.NETCoreApp,Version=v6.0.AssemblyAttributes.cs(2,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\LondonTravel.Skill.EndToEndTests.csproj]
C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\obj\Debug\net6.0\LondonTravel.Skill.EndToEndTests.AssemblyInfo.cs(11,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\LondonTravel.Skill.EndToEndTests.csproj]

Build FAILED.

C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\obj\Debug\net6.0\.NETCoreApp,Version=v6.0.AssemblyAttributes.cs(2,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\LondonTravel.Skill.csproj]
C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\obj\Debug\net6.0\LondonTravel.Skill.AssemblyInfo.cs(11,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\src\LondonTravel.Skill\LondonTravel.Skill.csproj]
C:\Users\martin.costello\.nuget\packages\microsoft.net.test.sdk\16.10.0\build\netcoreapp2.1\Microsoft.NET.Test.Sdk.Program.cs(2,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\LondonTravel.Skill.EndToEndTests.csproj]
C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\obj\Debug\net6.0\.NETCoreApp,Version=v6.0.AssemblyAttributes.cs(2,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\LondonTravel.Skill.EndToEndTests.csproj]
C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\obj\Debug\net6.0\LondonTravel.Skill.EndToEndTests.AssemblyInfo.cs(11,7): error CS0105: The using directive for 'System' appeared previously in this namespace [C:\Coding\martincostello\alexa-london-travel\test\LondonTravel.Skill.EndToEndTests\LondonTravel.Skill.EndToEndTests.csproj]
    0 Warning(s)
    5 Error(s)

Time Elapsed 00:00:03.01
```

Generated code is in the section below.

<details>

`src\LondonTravel.Skill\obj\Debug\net6.0\.NETCoreApp,Version=v6.0.AssemblyAttributes.cs`

```csharp
// <autogenerated />
using System;
using System.Reflection;
[assembly: global::System.Runtime.Versioning.TargetFrameworkAttribute(".NETCoreApp,Version=v6.0", FrameworkDisplayName = "")]

```

`src\LondonTravel.Skill\obj\Debug\net6.0\LondonTravel.Skill.AssemblyInfo.cs`

```csharp
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: System.Reflection.AssemblyCompanyAttribute("https://github.com/martincostello/alexa-london-travel")]
[assembly: System.Reflection.AssemblyConfigurationAttribute("Debug")]
[assembly: System.Reflection.AssemblyCopyrightAttribute("Martin Costello (c) 2017-2021")]
[assembly: System.Reflection.AssemblyFileVersionAttribute("2.0.4.0")]
[assembly: System.Reflection.AssemblyInformationalVersionAttribute("2.0.4+9d7343bedcc72cd6e5a215109430811507b0bf04")]
[assembly: System.Reflection.AssemblyProductAttribute("LondonTravel.Skill")]
[assembly: System.Reflection.AssemblyTitleAttribute("LondonTravel.Skill")]
[assembly: System.Reflection.AssemblyVersionAttribute("2.0.4.0")]
[assembly: System.Reflection.AssemblyMetadataAttribute("RepositoryUrl", "https://github.com/martincostello/alexa-london-travel.git")]
[assembly: System.Resources.NeutralResourcesLanguageAttribute("en-US")]

// Generated by the MSBuild WriteCodeFragment class.


```

`test\LondonTravel.Skill.EndToEndTests\obj\Debug\net6.0\.NETCoreApp,Version=v6.0.AssemblyAttributes.cs`

```csharp
// <autogenerated />
using System;
using System.Reflection;
[assembly: global::System.Runtime.Versioning.TargetFrameworkAttribute(".NETCoreApp,Version=v6.0", FrameworkDisplayName = "")]

```

`test\LondonTravel.Skill.EndToEndTests\obj\Debug\net6.0\LondonTravel.Skill.EndToEndTests.AssemblyInfo.cs`

```csharp
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: System.Reflection.AssemblyCompanyAttribute("https://github.com/martincostello/alexa-london-travel")]
[assembly: System.Reflection.AssemblyConfigurationAttribute("Debug")]
[assembly: System.Reflection.AssemblyCopyrightAttribute("Martin Costello (c) 2017-2021")]
[assembly: System.Reflection.AssemblyFileVersionAttribute("2.0.4.0")]
[assembly: System.Reflection.AssemblyInformationalVersionAttribute("2.0.4+9d7343bedcc72cd6e5a215109430811507b0bf04")]
[assembly: System.Reflection.AssemblyProductAttribute("LondonTravel.Skill.EndToEndTests")]
[assembly: System.Reflection.AssemblyTitleAttribute("LondonTravel.Skill.EndToEndTests")]
[assembly: System.Reflection.AssemblyVersionAttribute("2.0.4.0")]
[assembly: System.Reflection.AssemblyMetadataAttribute("RepositoryUrl", "https://github.com/martincostello/alexa-london-travel.git")]
[assembly: System.Resources.NeutralResourcesLanguageAttribute("en-US")]

// Generated by the MSBuild WriteCodeFragment class.


```

`C:\Users\martin.costello\.nuget\packages\microsoft.net.test.sdk\16.10.0\build\netcoreapp2.1\Microsoft.NET.Test.Sdk.Program.cs`

```csharp
// <auto-generated> This file has been auto generated. </auto-generated>
using System;
[Microsoft.VisualStudio.TestPlatform.TestSDKAutoGeneratedCode]
class AutoGeneratedProgram {static void Main(string[] args){}}
```

`test\LondonTravel.Skill.Tests\obj\Debug\net6.0\.NETCoreApp,Version=v6.0.AssemblyAttributes.cs`

```csharp
// <autogenerated />
using System;
using System.Reflection;
[assembly: global::System.Runtime.Versioning.TargetFrameworkAttribute(".NETCoreApp,Version=v6.0", FrameworkDisplayName = "")]

```

`test\LondonTravel.Skill.Tests\obj\Debug\net6.0\LondonTravel.Skill.Tests.AssemblyInfo.cs`

```csharp
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: System.Reflection.AssemblyCompanyAttribute("https://github.com/martincostello/alexa-london-travel")]
[assembly: System.Reflection.AssemblyConfigurationAttribute("Debug")]
[assembly: System.Reflection.AssemblyCopyrightAttribute("Martin Costello (c) 2017-2021")]
[assembly: System.Reflection.AssemblyFileVersionAttribute("2.0.4.0")]
[assembly: System.Reflection.AssemblyInformationalVersionAttribute("2.0.4+9d7343bedcc72cd6e5a215109430811507b0bf04")]
[assembly: System.Reflection.AssemblyProductAttribute("LondonTravel.Skill.Tests")]
[assembly: System.Reflection.AssemblyTitleAttribute("LondonTravel.Skill.Tests")]
[assembly: System.Reflection.AssemblyVersionAttribute("2.0.4.0")]
[assembly: System.Reflection.AssemblyMetadataAttribute("RepositoryUrl", "https://github.com/martincostello/alexa-london-travel.git")]
[assembly: System.Resources.NeutralResourcesLanguageAttribute("en-US")]

// Generated by the MSBuild WriteCodeFragment class.


```

</details>

**Comment by Youssef1313** [comment]
I don't think any compiler diagnostics should be implicitly suppressed for auto-generated files (this is different than code-style rules). Instead, the tool that generates the file should be fixed.

**Comment by martincostello** [comment]
I think that's fair, but as this seems to be coming from multiple MSBuild/compiler/SDK related code generators it was best to log the issue here to start with as a new language feature is the catalyst for the problem.

**Comment by PathogenDavid** [comment]
It's already the case that the global `<Nullable>` setting doesn't apply to generated files and there's a different set of nullable warnings which indicate as such, so there is some precedent that they're treated differently.

My impulse is that maybe global usings shouldn't even apply to generated code. They're a feature to make human lives easier, and they'd really only be able to benefit project-specific generators. Properly handling them in general-purpose generators sounds tedious and annoying.

**Comment by martincostello** [comment]
Related: https://github.com/dotnet/sdk/issues/18345

**Comment by PathogenDavid** [comment]
Earlier this week LDM [made the decision](https://github.com/dotnet/csharplang/blob/70fc3eb70379d5618ca9d2ae2dc897e02b92b678/meetings/2021/LDM-2021-06-21.md#email-decision-duplicate-global-using-warnings) to remove duplicate using warnings caused by global using statements.

**Comment by jaredpar** [comment]
This is fixed by #54220
